packages/flame/lib/src/components/core/component.dart

```diff
@@ -619,6 +619,24 @@ class Component {
     parent = newParent;
   }
 
+  /// Replaces a [replaceFrom] child of this component with [replaceWith].
+  /// I.e., removes [replaceFrom], and adds [replaceWith] instead.
+  FutureOr<void> replaceChild(Component replaceFrom, Component replaceWith) {
+    if (replaceFrom == replaceWith) {
+      // nothing to do
+      return Future.value();
+    }
```

Comment:
could just allow it to trigger any lifecycle hooks
or could throw an error

Reply:
If there is nothing to do, you can simply `return;`

Reply:
compiler wasn't happy; let me try again